File: avaliacao/src/main/java/br/com/zetta/service/CargoService.java
package br.com.zetta.service;

import br.com.zetta.service.dto.CargoDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface CargoService {

    List<CargoDto> findAll();

    Page<CargoDto> findByNome(String nome, Pageable pageable);

    CargoDto findById(Long id);

    CargoDto save(CargoDto cargoDto);

    CargoDto update(Long id, CargoDto cargoDto);

    void delete(Long id);
}

File: zetta-app/src/app/service/cargo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Cargo } from '../model/cargo';
import { Observable } from 'rxjs';
import { Pageable } from '../util/pageable';
import { RequestUtil } from '../shared/request-util';

@Injectable({
  providedIn: 'root'
})
export class CargoService {

  private cargosUrl: string;

  constructor(private http: HttpClient) {
    this.cargosUrl = 'http://localhost:8080/api/cargo';
  }

  findAll(): Observable<Cargo[]> {
    return this.http.get<Cargo[]>(this.cargosUrl);
  }

  findByNome(nome: string, pageable?: Pageable): Observable<any> {
    let params = RequestUtil.getRequestParams( pageable )
    params = params.append('nome', nome);

    return this.http.get<any>(`${this.cargosUrl}/filtrar`, { params: params });
  }

  findById(id: number): Observable<Cargo> {
    return this.http.get<Cargo>(`${this.cargosUrl}/${id}`);
  }

  save(cargo: Cargo): Observable<Cargo> {
    return this.http.post<Cargo>(this.cargosUrl, cargo);
  }

  update(id: number, cargo: Cargo): Observable<Cargo> {
    return this.http.put<Cargo>(`${this.cargosUrl}/${id}`, cargo);
  }

  delete(id: number): Observable<any> {
    return this.http.delete(`${this.cargosUrl}/${id}`);
  }
}
